src/details/ArborX_DetailsPermutedData.hpp

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILS_PERMUTED_DATA_HPP
+#define ARBORX_DETAILS_PERMUTED_DATA_HPP
+
+#include <ArborX_AccessTraits.hpp>
+
+namespace ArborX
+{
+
+namespace Details
+{
+
+template <typename Data, typename Permute, bool AttachIndices = false>
+struct PermutedData
+{
+  Data _data;
+  Permute _permute;
+  KOKKOS_FUNCTION auto &operator()(int i) const { return _data(_permute(i)); }
+};
+
+} // namespace Details
+
+template <typename Predicates, typename Permute, bool AttachIndices>
+struct AccessTraits<Details::PermutedData<Predicates, Permute, AttachIndices>,
+                    PredicatesTag>
+{
+  using PermutedPredicates =
+      Details::PermutedData<Predicates, Permute, AttachIndices>;
+  using NativeAccess = AccessTraits<Predicates, PredicatesTag>;
+
+  static std::size_t size(PermutedPredicates const &permuted_predicates)
+  {
+    return NativeAccess::size(permuted_predicates._data);
+  }
+
+  template <bool _Attach = AttachIndices>
```

Comment:
No do not prepend `_`

Reply:
I did not touch this code, it was taken straight out of `ArborX_DetailsBufferOptimization.hpp`. I did not even look at it.

Reply:
OK